internal/command/command.go

```diff
@@ -0,0 +1,307 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/psiemens/sconfig"
+
+	"github.com/onflow/flow-cli/pkg/flow"
+	"github.com/onflow/flow-cli/pkg/flow/gateway"
+	"github.com/onflow/flow-cli/pkg/flow/services"
+	"github.com/onflow/flow-cli/pkg/flow/util"
+	"github.com/onflow/flow-go-sdk/client"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type RunCommand func(
+	*cobra.Command,
+	[]string,
+	*flow.Project,
+	*services.Services,
+) (Result, error)
+
+type Command struct {
+	Cmd   *cobra.Command
+	Flags interface{}
+	Run   RunCommand
+}
+
+var (
+	FilterFlag      = ""
+	FormatFlag      = ""
+	SaveFlag        = ""
+	RunEmulatorFlag = false
+	HostFlag        = ""
+	LogFlag         = "info"
+	NetworkFlag     = ""
+)
+
+// InitFlags init all the global persistent flags
+func InitFlags(cmd cobra.Command) {
+	cmd.PersistentFlags().StringVarP(
+		&FilterFlag,
+		"filter",
+		"x",
+		FilterFlag,
+		"Filter result values by property name",
+	)
+
+	cmd.PersistentFlags().StringVarP(
+		&HostFlag,
+		"host",
+		"",
+		HostFlag,
+		"Flow Access API host address",
+	)
+
+	cmd.PersistentFlags().StringVarP(
+		&FormatFlag,
+		"output",
+		"o",
+		FormatFlag,
+		"Output format, values (json, inline)",
+	)
+
+	cmd.PersistentFlags().StringVarP(
+		&SaveFlag,
+		"save",
+		"s",
+		SaveFlag,
+		"Save result to a filename",
+	)
+
+	cmd.PersistentFlags().StringVarP(
+		&LogFlag,
+		"log",
+		"l",
+		LogFlag,
+		"Log level verbosity, values (none, error, debug)",
+	)
+
+	cmd.PersistentFlags().BoolVarP(
+		&RunEmulatorFlag,
+		"emulator",
+		"e",
+		RunEmulatorFlag,
+		"Run in-memory emulator",
+	)
+
+	cmd.PersistentFlags().StringSliceVarP(
+		&flow.ConfigPath,
+		"conf",
+		"f",
+		flow.ConfigPath,
+		"Path to flow configuration file",
+	)
+
+	cmd.PersistentFlags().StringVarP(
+		&NetworkFlag,
+		"network",
+		"n",
+		NetworkFlag,
+		"Network from configuration file",
+	)
+}
+
+// addCommand add new command to main cmd
+// and initializes all necessary things as well as take care of errors and output
+// here we can do all boilerplate code that is else copied in each command and make sure
+// we have one place to handle all errors and ensure commands have consistent results
+func (c Command) Add(parent *cobra.Command) {
```

Comment:
```suggestion
func (c Command) AddToParent(parent *cobra.Command) {
```

The naming confused me a bit, then I realized that it works backwards (child adds itself to parent rather than parent adds child)